app/views/core/CreateAccountModal/teacher/SetupAccountPanel.coffee

```diff
@@ -19,10 +19,24 @@ SetupAccountPanel = Vue.extend
       @saving = false
   methods:
     clickFinish: ->
-      # Make sure to add conditions if we change this to be used on non-teacher path
-      window.tracker?.trackEvent 'CreateAccountModal Teacher SetupAccountPanel Finish Clicked', category: 'Teachers'
-      application.router.navigate('teachers/classes', {trigger: true})
-      document.location.reload()
+      # Save annoucements subscribe info
+      me.fetch(cache: false)
+      .then =>
+        emails = _.assign({}, me.get('emails'))
+        emails.generalNews ?= {}
+        emails.generalNews.enabled = $('#subscribe-input').is(':checked')
```

Comment:
This works but goes against the vue-y patterns elsewhere of keeping the state in the component's state object.

Reply:
Do you want to submit an update to change this before I commit to changes?